### DISCUSSION TITLE
Hydrogen would be great if you could do this.

### DISCUSSION THREAD

**Discussion by IV3RSONCL** [hydrogen-music/hydrogen#1780] [Hydrogen on Windows]
I am an old user of hydrogen, sorry for my English, I went through many phases of this program, I am not a producer, I am currently more of a singer, but I started making incredible beats of trap reggaeton, etc, etc. It is a great program that if it continued to exploit, it would be the king without a doubt. you can create beats very fast

///////// well the important thing ///////
I would like to know if 2 small squares can be added in a song pattern to create a melody underneath. I understand that many will say it is better to use a daw and take out the drum, but I only know how to use hydrogen because of my little knowledge of daws, so yes, it could be done This would be interesting at least if you are not convinced to create the melody and then create it in fl studio or whatever. I also understand that many melodies are not embedded in the percussion but what do you think? Could you tell me if this can be done?

![imagen](https://github.com/hydrogen-music/hydrogen/assets/134195226/baf92379-71dd-4c2e-a7f3-653930c1bfdf)

**Comment by theGreatWhiteShark** [comment]
> I would like to know if 2 small squares can be added in a song pattern to create a melody underneath.

No. A grid cell is still the basic unit of the song editor and you can not toggle one or multiple subsets.

We actually get questions like this a lot recently. Most probably because of the rendering of shorter patterns as partially filled squares. The whole idea behind it was to visually indicate differences in length between the patterns. E.g. when adding a pattern to a column which is twice as long as all the other present ones, you immediately see that there is a larger silence introduce after them.

But I get what you want and I see that it is quite useful. However, the concept of a grid with at most one cell per row and column is baked into Hydrogen. Coming up with a concept and implementation to allow cells span multiple columns or having several ones start within the same will be tremendous change and quite some work to deal with all the side effects and to integrate the UX seamlessly into the existing one.

I don't say no and wouldn't say never, but for now there is neither a plan nor sufficient resources on our development team to take such step.

The workaround for now is still to make the smallest pattern one grid cell long and divide the larger one into multiple parts you always activate together.
